### ISSUE TITLE
Alter type for enum

### ISSUE THREAD

**Issue by tyt2y3** [SeaQL/sea-query#38]
Follow up on #2, where `CREATE TYPE` and `DROP TYPE` for ENUMs was already implemented in 
https://github.com/SeaQL/sea-query/blob/master/src/extension/postgres/types.rs 
We now want to implement `ALTER TYPE ADD VALUE` and `ALTER TYPE RENAME VALUE` etc to complete the use case.
And prepare testcases for them.

**Comment by MozarellaMan** [comment]
Hi! I'd like to try this. We've talked on email before, sorry had exams the past week. Just one question, in the tests directory, is it okay to have an `use` statement in a new `tests/postgres/types.rs` file? It seems like `sea-query-test` cannot see the Type module in `postgress/types.rs`. This is so I can add tests

**Comment by tyt2y3** [comment]
Hi welcome! Yes sure, I think extension::* were not visible by default, as they are optionals.
I actually put two type test cases in https://github.com/SeaQL/sea-query/blob/master/tests/postgres/table.rs#L113
I guess it is good to move them out into a new file now.

**Comment by billy1624** [comment]
Should we also implement rename existing enum value?

```sql
ALTER TYPE name RENAME VALUE existing_enum_value TO new_enum_value
```

https://www.postgresql.org/docs/current/sql-altertype.html

**Comment by tyt2y3** [comment]
For the scope of this PR, I think we only needed to support `ALTER TYPE name RENAME TO new_name` (done), `ALTER TYPE name ADD VALUE [ IF NOT EXISTS ] new_enum_value [ { BEFORE | AFTER } neighbor_enum_value ]` (done) and
`ALTER TYPE name RENAME VALUE existing_enum_value TO new_enum_value` since others are not related to enums.

All in all thank you for the contribution!

**Comment by tyt2y3** [comment]
Included in `0.10.6`
